fenix/app/src/main/java/org/mozilla/fenix/browser/tabstrip/TabsStrip.kt

```diff
@@ -0,0 +1,371 @@
+    onSelectedTabClick: (id: String) -> Unit,
+) {
+    BoxWithConstraints(modifier = modifier) {
+        val listState = rememberLazyListState()
+        val tabWidth = ((maxWidth - tabStripStartPadding) / state.tabs.size) - spaceBetweenTabs
+
+        LazyRow(
+            modifier = Modifier,
+            state = listState,
+            contentPadding = PaddingValues(start = tabStripStartPadding),
+        ) {
+            items(
+                items = state.tabs,
+                key = { it.id },
+            ) { itemState ->
+                TabItem(
+                    state = itemState,
+                    onCloseTabClick = onCloseTabClick,
+                    onSelectedTabClick = onSelectedTabClick,
+                    modifier = Modifier
+                        .animateItemPlacement()
+                        .width(
+                            tabWidth.coerceIn(
+                                minimumValue = minTabStripItemWidth,
+                                maximumValue = maxTabStripItemWidth,
+                            ),
+                        ),
+                )
+
+                Spacer(modifier = Modifier.size(spaceBetweenTabs))
+            }
+        }
+
+        if (state.tabs.isNotEmpty()) {
+            // When a new tab is added, scroll to the end of the list
+            LaunchedEffect(state.tabs.last().id) {
+                listState.scrollToItem(state.tabs.size)
+            }
+
+            // When a tab is selected, scroll to the selected tab
+            val selectedTab = state.tabs.firstOrNull { it.isSelected }
+            LaunchedEffect(selectedTab?.id) {
+                if (selectedTab != null) {
+                    val selectedItemInfo =
+                        listState.layoutInfo.visibleItemsInfo.firstOrNull { it.key == selectedTab.id }
+
+                    if (listState.isItemPartiallyVisible(selectedItemInfo) || selectedItemInfo == null) {
+                        listState.animateScrollToItem(state.tabs.indexOf(selectedTab))
+                    }
+                }
+            }
+        }
+    }
+}
+
+private fun LazyListState.isItemPartiallyVisible(itemInfo: LazyListItemInfo?) =
+    itemInfo != null &&
+        (itemInfo.offset + itemInfo.size > layoutInfo.viewportEndOffset || itemInfo.offset < 0)
+
+@Composable
+private fun TabItem(
+    state: TabStripItem,
+    modifier: Modifier = Modifier,
+    onCloseTabClick: (id: String) -> Unit,
+    onSelectedTabClick: (id: String) -> Unit,
+) {
+    TabStripCard(
+        modifier = modifier.fillMaxSize(),
+        backgroundColor =
+        if (state.isPrivate) {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer3
+            } else {
+                FirefoxTheme.colors.layer2
+            }
+        } else {
+            if (state.isSelected) {
+                FirefoxTheme.colors.layer2
+            } else {
+                FirefoxTheme.colors.layer3
+            }
+        },
```

Comment:
a `when` might help visually clean this up

Reply:
could even extract this block of logic to a func altogether, such as `getBackgroudColor`

Reply:
I think if reads better here since there are two nested conditions, I will wait to get the final background colors before extracting it out in a function, think the updated colors might just simplify the logic.